Modules/_testinternalcapi.c

```diff
@@ -102,6 +102,43 @@ test_popcount(PyObject *self, PyObject *Py_UNUSED(args))
 }
 
 
+static int
+check_bit_length(unsigned long x, int expected)
+{
+    int len = _Py_bit_length(x);
```

Comment:
I suggest volatile here. For check_popcount(), using GCC -O3 without volatile: all calls to check_popcount() have been removed by the compiler which considers that the tests are correct at built time and so can be removed. See check_popcount() comment.